### PR TITLE
Update Membership Routes

### PR DESCRIPTION
**Pull request by Abban**
Integrate payment context changes into this
controller and presenter

Ticket: https://phabricator.wikimedia.org/T309088

### CODE REVIEW

File: tests/EdgeToEdge/Routes/ApplyForMembershipRouteTest.php [hunk 1]
```diff
@@ -620,4 +528,8 @@ private function insertIncentives( FunFunFactory $factory, Incentive ...$incenti
 		}
 		$em->flush();
 	}
+
+	private function insertPaymentId( FunFunFactory $factory, int $index ): void {
```
Comment (gbirke) [inline tests/EdgeToEdge/Routes/ApplyForMembershipRouteTest.php:532]: Manually inserting a payment ID should no longer be necessary, it's part of the test environment setup in `tests/RebuildDatabaseSchemaTrait.php`

**Comment by gbirke** [comment]
There were some recent payment changes (checking allowed payment types) that might affect how the controllers for applying for membership and validating the fee construct classes. Please review https://github.com/wmde/fundraising-application/pull/2515 and merge it, rebasing this PR on top of it

File: tests/EdgeToEdge/Routes/ApplyForMembershipRouteTest.php (hunk 1 above)
Comment (Abban) [reply tests/EdgeToEdge/Routes/ApplyForMembershipRouteTest.php:532]: Ah great, I was waiting for a standup to ask about the best way to do this with the test setup.

File: tests/EdgeToEdge/Routes/ApplyForMembershipRouteTest.php [hunk 2]
```diff
@@ -68,17 +69,15 @@ public function testGivenGetRequestSustainingMembership_formIsShown(): void {
 	}
 
 	public function testGivenRequestWithDonationIdAndCorrespondingAccessCode_successResponseWithInitialFormValuesIsReturned(): void {
-		$this->markTestIncomplete( "This should work again when we finish updating the membership controllers" );
+		$this->markTestIncomplete( "This should work again when we finish updating the donation controllers" );
```
Comment (gbirke) [inline tests/EdgeToEdge/Routes/ApplyForMembershipRouteTest.php:72]: Changing `DonationMembershipApplicationAdapter` and making this test pass should be part of this ticket. The `update-payment-context` branch has all the necessary donation changes.

File: tests/EdgeToEdge/Routes/ApplyForMembershipRouteTest.php [hunk 3]
```diff
@@ -361,40 +340,19 @@ public function testWhenMultipleMembershipInAccordanceToTimeLimit_isNotRejected(
...
-	public function testGivenValidRequestUsingPayPal_applicationIsPersisted(): void {
-		$this->markTestIncomplete( "This should work again when we finish updating the membership controllers" );
-
-		$this->createEnvironment( function ( Client $client, FunFunFactory $factory ): void {
-			$factory->setPaymentDelayCalculator( $this->newFixedPaymentDelayCalculator() );
-
-			$client->request(
-				'POST',
-				'/apply-for-membership',
-				$this->newValidHttpParametersUsingPayPal()
```
Comment (gbirke) [inline tests/EdgeToEdge/Routes/ApplyForMembershipRouteTest.php:380]: This method can now also be removed

**Review by gbirke — CHANGES_REQUESTED** [review]

**Review by gbirke — APPROVED** [review]
Everything works now except the small thing I corrected with #2524

**Pull request merged**